File: python/AccessTokenTest.py
# cd python/src && python AccessTokenTest.py
import sys
import os
import time

sys.path.append(os.path.join(os.path.dirname(__file__), '../src'))
import AccessToken

app_id = "123456781234567812345678"
app_key = "app key"
room_id = "new room"
user_id = "new user id"

token = AccessToken.AccessToken(app_id, app_key, room_id, user_id)
token.add_privilege(AccessToken.PrivSubscribeStream, 0)
token.add_privilege(AccessToken.PrivPublishStream, int(time.time()) + 3600)
token.expire_time(int(time.time()) + 3600)

s = token.serialize()

print(s)

t = AccessToken.parse(s)

print(t.verify(app_key))

File: php/AccessTokenTest.php
// cd php/src && php AccessToken.php
<?php
include "../src/AccessToken.php";

date_default_timezone_set("UTC");
$date = new DateTime();

$appID = "123456781234567812345678";
$appKey = "app key";
$roomID = "new room";
$userID = "new user id";

$token = AccessToken::init($appID, $appKey, $roomID, $userID);

$token->expireTime(0);
$token->addPrivilege($Privileges["PrivSubscribeStream"], 0);
$token->addPrivilege($Privileges["PrivPublishStream"], $date->getTimestamp() + 3600);

$s = $token->serialize();

var_dump($s);


$t = AccessToken::parse($s);

var_dump($t);

var_dump($t->verify($appKey));
?>